app/presenters/menu/settings_loader.rb

```diff
@@ -0,0 +1,36 @@
+module Menu
+  class SettingsLoader < YamlLoader
+    include Singleton
+
+    def self.load
+      instance.load_from_settings
+    end
+
+    def load_from_settings
+      begin
+        settings = ::Settings.ui.custom_menu
+        items = (settings || []).map { |i| create_custom_item(HashWithIndifferentAccess.new(i)) }
```

Comment:
What happens if there is bad configuration?  Will it blow up?  I'm not sure what create_custom_menu_item does under the covers.

Reply:
The idea is that if there's a bad configuration something would get logged and the settings will be ignored.
Do you see any problem with this approach?